features/ui/src/main/java/org/opennms/alec/grpc/SituationClient.java

```diff
@@ -28,34 +28,72 @@
 
 package org.opennms.alec.grpc;
 
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
 import org.opennms.alec.datasource.api.Situation;
 import org.opennms.alec.grpc.generated.AlecCollectionServiceGrpc;
 import org.opennms.alec.grpc.generated.SituationSet;
 import org.opennms.alec.mapper.SituationToSituationProto;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import io.grpc.Channel;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.google.protobuf.Empty;
+
+import io.grpc.ManagedChannel;
+import io.grpc.ManagedChannelBuilder;
 
 public class SituationClient {
     private static final Logger LOG = LoggerFactory.getLogger(SituationClient.class);
 
-    private final AlecCollectionServiceGrpc.AlecCollectionServiceBlockingStub blockingStub;
+    private final AlecCollectionServiceGrpc.AlecCollectionServiceFutureStub futureStub;
 
     private final SituationToSituationProto mapper;
     private final boolean doStore;
+    private final ManagedChannel channel;
+    private final ExecutorService executorService = Executors.newSingleThreadExecutor();
 
-    public SituationClient(Channel channel, SituationToSituationProto mapper, boolean doStore) {
-        blockingStub = AlecCollectionServiceGrpc.newBlockingStub(channel);
+    public SituationClient(SituationToSituationProto mapper, boolean doStore, GrpcConnectionConfig grpcConnectionConfig) {
+        channel = ManagedChannelBuilder
+                .forAddress(grpcConnectionConfig.getHost(), grpcConnectionConfig.getPort())
+                .build();
```

Comment:
Does this initiate the connection to the server, or does it only prepare the channel?

We don't want any outbound connections to be attempted, unless the user has explicitly opted in.